NuRadioMC/simulation/simulation.py

```diff
@@ -432,7 +432,10 @@ def run(self):
             t1 = time.time()
             iE_mother = event_indices[0]
             x_int_mother = np.array([self._fin['xx'][iE_mother], self._fin['yy'][iE_mother], self._fin['zz'][iE_mother]])
-            self._mout['weights'][event_indices] = get_weight(self._fin['zeniths'][iE_mother],
+            if("weights" in self._fin):
+                self._mout['weights'] = self._fin["weights"]
+            else:
```

Comment:
In the past I have looked at the attrs of a hdf5 file to see which weights are in the file. Is it still guaranteed that the  information written in `self._cfg['weights']['weight_mode']` and `self._cfg['weights']['cross_section_type']` will be correct?

Would be annoying if one needs to reverse engineer and verify what has really been run by re-calculating event weights by hand... ;)

Reply:
looks better to me